_layouts/team-member.html

```diff
@@ -48,20 +48,20 @@ <h1 class="title">{{ page.firstname }} {{ page.lastname }}
             {% endif %}
         </div>
     </div>
+</div>
 
-    <!-- blog posts -->
-    <div class="team-posts">
-    {% assign has-title = false %}
-
-    {% for post in site.posts %}
-        {% for author in post.authors %}
-        {% if author.fname == page.firstname and author.lname == page.lastname %}
-
-        {% unless has-title %}
-        <h2 class="title"> Blog posts that {{ page.firstname }} has written: </h2>
-        {% assign has-title = true %}
-        {% endunless %}
-
+<!-- blog posts -->
+<div class="team-posts">
+    {% assign authors_posts = site.posts | where_exp:"post", "post.authors contains page.name" | sort_by:"date" %}
+    {% if authors_posts.size > 0 %}
+        {% assign firstname = page.name | split: " " | first %}
+        {% if page.pronouns and page.pronouns contains "they" %}
+            {% assign have_has = "have" %}
+        {% else %}
+            {% assign have_has = "has" %}
+        {% endif %}
+        <h2 class="title"> Blog posts that {{ firstname }} {{ have_has }} written: </h2>
```

Comment:
I think this is my only nitpick; regardless of pronouns (e.g. they/them), I think we'd still use has.

For example:
* Blog posts that Matt has written (he/him)
* Blog posts that Sophie has written (she/her)
* Blog posts that Arjun has written (they/them)

So, no need for the `have_has` variable, though I appreciate you thinking about this!